libvast/src/system/index.cpp

```diff
@@ -1968,10 +1972,11 @@ index(index_actor::stateful_pointer<index_state> self,
                               });
                         }
                       },
-                      [self](const caf::error& e) {
+                      [self, rp](caf::error e) mutable {
```

Comment:
```suggestion
                      [self, rp](caf::error& e) mutable {
```

This actually took the lvalue-ref intentionally, see https://github.com/actor-framework/actor-framework/discussions/1220 for why.

Reply:
Thanks for the clarification